src/macros.cr

```diff
@@ -61,7 +61,17 @@
 # p.copy_with x: 3   # => #<Point(@x=3, @y=2)>
 # p                  # => #<Point(@x=0, @y=2)>
 # ```
-macro record(name, *properties)
+macro record(name, *properties, **kwargs)
+  {% raise "#{name} does not accept named arguments
+    You probably wrote:
+
+      record #{name}, name: String
+
+    instead of:
+
+      record #{name}, name : String
```

Comment:
We know exactly how the code looks like at the call site. But more importantly, we can make a good guess about the intention.
So I'd suggest implementing an informed "Did you mean?" suggestion:
```cr
<<-TXT
Did you mean this?

  record #{name}, #{(properties + kwargs.map{|name, value| "#{name} : #{value}"}).join(", ").id}
TXT
```